File: app/src/main/java/com/example/santanderapp/santander/adapter/StatementAdapter.java
package com.example.santanderapp.santander.adapter;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.example.santanderapp.santander.R;
import com.example.santanderapp.santander.model.StatementList;

import java.text.DateFormat;
import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.List;
import java.util.Locale;

import static com.example.santanderapp.santander.util.Utils.convertData;

public class StatementAdapter extends RecyclerView.Adapter<StatementAdapter.MeuViewHolder> {

    private List<StatementList> ListStatement;
    private SimpleDateFormat formatIso;
    private SimpleDateFormat formatBra;
    private Date date;

    public StatementAdapter(List<StatementList> statement) {
        this.ListStatement = statement;
    }

    @Override
    public MeuViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.card_view_item, parent, false);

        return new MeuViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(MeuViewHolder holder, int position) {
        StatementList item = ListStatement.get(position);
        holder.typeAccount.setText(String.valueOf(item.title));
        try {
            holder.dateAccount.setText(convertData(String.valueOf(item.date)));
        } catch (ParseException e) {
            e.printStackTrace();
        }
        holder.descriptionAccount.setText(String.valueOf(item.desc));
        holder.valueAccount.setText(String.valueOf(item.value));
    }

    @Override
    public int getItemCount() {
        return ListStatement.size();
    }



    public class MeuViewHolder extends RecyclerView.ViewHolder {
        public TextView typeAccount, dateAccount, descriptionAccount, valueAccount;

        public MeuViewHolder(View view) {
            super(view);
            typeAccount = (TextView) view.findViewById(R.id.typeAccount);
            dateAccount = (TextView) view.findViewById(R.id.dateAccount);
            descriptionAccount = (TextView) view.findViewById(R.id.descriptionAccount);
            valueAccount = (TextView) view.findViewById(R.id.valueAccount);
        }
    }

}

File: app/src/main/java/com/example/testeandroidsantander/recyclerview/StatementAdapter.kt
package com.example.testeandroidsantander.recyclerview

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import com.example.testeandroidsantander.R
import com.example.testeandroidsantander.model.Statement
import com.example.testeandroidsantander.model.StatementList
import kotlinx.android.synthetic.main.bank_statement_item.view.*
import java.text.DecimalFormat


class StatementAdapter(private val statement: StatementList,
                       private val context: Context
) : RecyclerView.Adapter<StatementAdapter.ViewHolder>() {

    override fun getItemCount(): Int {
        return statement.statementList.size
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val note = statement.statementList[position]
        holder.bindView(note)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(context).inflate(R.layout.bank_statement_item, parent, false)
        return ViewHolder(view)
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        val recentLabel: TextView
        val typeTextView: TextView
        val dateLabel: TextView
        val valueTextView: TextView
        var item: Statement? = null

        init {
            recentLabel = itemView.recentLabel
            typeTextView = itemView.typeTextView
            dateLabel = itemView.dateLabel
            valueTextView = itemView.valueTextView
        }

        fun bindView(item: Statement) {
            this.item = item

            val recentLabel = itemView.recentLabel
            val typeTextView = itemView.typeTextView
            val dateLabel = itemView.dateLabel
            val valueTextView = itemView.valueTextView

            recentLabel.text = item.title
            typeTextView.text = item.desc
            dateLabel.text = item.date
            valueTextView.text = item.value
        }
    }
}